viewing_party/party.py

```diff
@@ -1,23 +1,156 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if (not title) or (not genre) or (not rating):
+        return None
+    
+    movie_dict = {"title": title, "genre": genre, "rating": rating}
+    return movie_dict
+
+def add_to_watched(user_data, movie):
+    new_watched_list = user_data["watched"]
+    new_watched_list.append(movie)
+    user_data["watched"] = new_watched_list
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    new_watchlist_list = user_data["watchlist"]
+    new_watchlist_list.append(movie)
+    user_data["watchlist"] = new_watchlist_list
+    return user_data
+
+def watch_movie(user_data, title): 
+    for movie in list(user_data["watchlist"]):
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    watched_movies_list = user_data["watched"]
+    if len(watched_movies_list) == 0:
+        return 0.0
+    sum_of_ratings = 0
+    for movie in watched_movies_list:
+        movie_rating = movie["rating"]
+        sum_of_ratings+= movie_rating

```

Comment:
we should combine this into one line:

```suggestion
        sum_of_rating += movie["rating"]

```
while it does the same thing, we can cut down on variables using one line. And I think it is easier for other coders to see what you are accomplishing with one line rather than the two.